File: back-end/domain-modules/runs-orchestrators-domain/runs-orchestrators-models/src/main/java/com/maukaim/bulo/runs/orchestrators/data/definition/IoDependency.java
package com.maukaim.bulo.runs.orchestrators.data.definition;

import java.util.Set;

public class IoDependency {
    private final String inputId;
    private final Set<InputProvider> inputProviders;

    public IoDependency(String inputId, Set<InputProvider> inputProviders) {
        this.inputId = inputId;
        this.inputProviders = inputProviders;
    }

    public String getInputId() {
        return inputId;
    }

    public Set<InputProvider> getInputProviders() {
        return inputProviders;
    }

    @Override
    public String toString() {
        return "IoDependency{" +
                "inputId='" + inputId + '\'' +
                ", inputProviders=" + inputProviders +
                '}';
    }
}

File: front-end/lib/domains/flows/model/io_dependency.dart

import 'input_provider.dart';

class IoDependency {
  final String inputId;
  final List<InputProvider> inputProviders;

  IoDependency({required this.inputId, required this.inputProviders});

  factory IoDependency.fromJson(Map<String, dynamic> json) {
    return IoDependency(
      inputId: json['inputId'],
      inputProviders: [
        for (var x in json['inputProviders']) InputProvider.fromJson(x)
      ],
    );
  }
}